logbook-json/src/main/java/org/zalando/logbook/json/JacksonJsonFieldBodyFilter.java

```diff
@@ -47,17 +47,16 @@ public String filter(@Nullable final String contentType, final String body) {
 
     public String filter(final String body) {
         try {
-            final JsonParser parser = factory.createParser(body);
 
-            final CharArrayWriter writer = new CharArrayWriter(body.length() * 2); // rough estimate of final size
 
-            final JsonGenerator generator = factory.createGenerator(writer);
-            try {
-                while (true) {
-                    JsonToken nextToken = parser.nextToken();
-                    if (nextToken == null) {
-                        break;
-                    }
+            try (final JsonParser parser = factory.createParser(body);
+
+                 final CharArrayWriter  writer = new CharArrayWriter(body.length() * 2); // rough estimate of final size
+
+                 final JsonGenerator generator = factory.createGenerator(writer)){
```

Comment:
Shouldn't generator also be closed?

Reply:
Ah, I just noticed that it's inside the try block. The spacing confused me at first.